data/src/test/java/org/apache/iceberg/data/DataTest.java

```diff
@@ -59,6 +63,10 @@ protected boolean supportsDefaultValues() {
     return false;
   }
 
+  protected boolean supportsRequiredTypes() {
```

Comment:
not sure if the method name is specific enough? it can be interpreted if it supports required field properly. 

I guess it really meant `handleNullValueWriteProperlyForRequiredType`?

Reply:
Alternatively, we can remove this boolean flag method. just override the test method in the `orc` test code and modify the expected behavior.